### PR TITLE
 Fixes #24980 - enable structured formatters for refernces. 

### PR DESCRIPTION
**Pull request by rahulbajaj0509**

### CODE REVIEW

File: test/unit/output/formatters_test.rb [hunk 1]
```diff
@@ -57,94 +58,102 @@
...
-      formatter.format(reference_str_keys, {}).must_equal 'Server'
+    it 'formats name' do
+      formatter.format(reference_str_keys, {}).must_equal("#{reference_str_keys['name']}")
     end
 
-    it "can override name key" do
+    it 'can override name key' do
       options = {:name_key => :another_name}
-      formatter.format(reference_str_keys, options).must_equal 'SERVER'
+      formatter.format(reference_str_keys, options).must_equal("#{reference_str_keys['name']}")
```
[the remark this answers is not in the document]
Comment (mbacovsky) [reply test/unit/output/formatters_test.rb]: Either I don't understand the test or something is wrong. If I "fix" the test it starts to fail. Any idea why?

[the remark this answers is not in the document]
Comment (tstrachota) [reply test/unit/output/formatters_test.rb]: IMHO the options are wrong in this test. Formatters don't accept `:name_key` but `:display_field_key`. See:
https://github.com/theforeman/hammer-cli-foreman/blob/adcc94f44e42bc9df69358bc340fdd1819a68faf/lib/hammer_cli_foreman/output/formatters.rb#L9

[the remark this answers is not in the document]
Comment (mbacovsky) [reply test/unit/output/formatters_test.rb]: I see now, thanks @tstrachota!

[the remark this answers is not in the document]
Comment (rahulbajaj0509) [reply test/unit/output/formatters_test.rb]: Good catch @mbacovsky, sorry I somehow missed that. Thanks @tstrachota again :)

**Comment by tstrachota** [comment]
Good job @rahulbajaj0509, thanks!

**Pull request merged**